pkg/cached/cache.go

```diff
@@ -278,35 +289,41 @@ func (c *static[T]) Get() Result[T] {
 	return c.result
 }
 
-// Replaceable is a cache that carries the result even when the cache is
-// replaced. This is the type that should typically be stored in
-// structs.
-type Replaceable[T any] struct {
-	cache  atomic.Pointer[Data[T]]
-	result atomic.Pointer[Result[T]]
+// Atomic is a cache adapter that can be atomically replaced with a new one.
+type Atomic[T any] struct {
+	value atomic.Pointer[Data[T]]
 }
 
-// Get retrieves the data from the underlying source. [Replaceable]
-// implements the [Data] interface itself. This is a pass-through
-// that calls the most recent underlying cache. If the cache fails but
-// previously had returned a success, that success will be returned
-// instead. If the cache fails but we never returned a success, that
-// failure is returned.
-func (c *Replaceable[T]) Get() Result[T] {
-	result := (*c.cache.Load()).Get()
-
-	for {
-		cResult := c.result.Load()
-		if result.Err != nil && cResult != nil && cResult.Err == nil {
-			return *cResult
-		}
-		if c.result.CompareAndSwap(cResult, &result) {
-			return result
-		}
-	}
+func (x *Atomic[T]) Store(val Data[T]) *Atomic[T] { x.value.Store(&val); return x }
+func (x *Atomic[T]) Get() Result[T]               { return (*x.value.Load()).Get() }
+
+// NewLastSuccess creates a new LastSuccess cache.
+func NewLastSuccess[T any]() *LastSuccess[T] {
```

Comment:
This seems to follow a different conventions than the previous caches with using `New` as part of the constructor?

Reply:
updated.